app/controllers/actions/AuthAction.scala

```diff
@@ -136,4 +144,23 @@ class AuthAction @Inject()(val authConnector: AuthConnector,
       .flatMap(_.getIdentifier(agentReferenceNumberIdentifier))
       .map(e => Arn(e.value))
   }
+
+  private def withSuspendedCheck(body: Option[Boolean] => Future[Result])
+                         (implicit reads: Reads[Boolean], request: Request[_], hc: HeaderCarrier, ec: ExecutionContext, appConfig: AppConfig): Future[Result] = {
+    sessionCacheService.get[Boolean](SUSPENSION_STATUS).flatMap {
```

Comment:
The downside of using a session cache for this is that if there are multiple users from the same ARN, each one will trigger a request to ACA, whereas a local cache (like caffeine) is keyed on the ARN, so only 1 request per ARN. We can always change it if we see multiple concurrent users from same ARN so let's leave it for now?